src/routes/index.svelte

```diff
@@ -2,27 +2,41 @@
 	import MissionCard from '$lib/cards/MissionCard.svelte';
 	import HomeSearchBar from '$lib/home/HomeSearchBar.svelte';
 	import type { Mission } from '$lib/types';
-	import { onMount } from 'svelte';
-
+	import type { RepoModule } from '$lib/repo';
 
 	export let missions: Mission[];
 	export let missionCards: any = {};
+	export let modules: RepoModule[];
+
+	let render = false;
+	let searchBar: HomeSearchBar;
 
-	missions.sort((a, b) => (a.name.toLowerCase() > b.name.toLowerCase() ? 1 : -1));
+	function onChange() {
+		render = false;
+		missions = missions;	//signals svelte to rerender the bombs section
+		setTimeout(() => {
+			render = true;
+			missions = missions;	//signals svelte to rerender the bombs section
+			missions.forEach(m => {
+				missionCards[m.name] = document.getElementById('mission-' + m.name);
+			});
+			setTimeout(() => {searchBar.updateSearch();},1000);
+		}, 50);
+	}
 </script>
 
 <svelte:head>
 	<title>Challenge Bombs</title>
 </svelte:head>
 <h1 class="header">Challenge Bombs</h1>
-<HomeSearchBar missions={missions}
-	missionCards={missionCards}>
-</HomeSearchBar>
+<HomeSearchBar bind:this={searchBar} bind:missions={missions} bind:missionCards={missionCards} on:change={onChange} {modules}/>
 <div class="bombs">
 	{#each missions as mission, index}
```

Comment:
I believe if you used a [keyed each](https://svelte.dev/tutorial/keyed-each-blocks), it would render this correctly:
```suggestion
	{#each missions as mission, index (mission.name)}
```